crates/outbound-http/src/lib.rs

```diff
@@ -31,8 +33,19 @@ impl OutboundHttp {
     /// If `None` is passed, the guest module is not allowed to send the request.
     fn is_allowed(&self, url: &str) -> Result<bool, HttpError> {
         let url = Url::parse(url).map_err(|_| HttpError::InvalidUrl)?;
+
+        if self.is_same_origin(&url) {
+            return Ok(true);
+        }
+
         Ok(self.allowed_hosts.allow(&url))
     }
+
+    /// Check if the outbound request host and port are the same as origin
+    fn is_same_origin(&self, url: &Url) -> bool {
```

Comment:
Maybe 
```suggestion
    fn is_same_as_origin(&self, url: &Url) -> bool {
```

?  "Same origin" implies to me whether two things have the same origin, whereas you are interested in whether the _URL_ (the _destination_) is the same as the origin.